lib/dry/cli/inline.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'backports/2.5.0/module/define_method' if RUBY_VERSION < '2.5'
+
+module Dry
+  class CLI
+    require 'dry/cli'
+
+    module Inline
+      extend Forwardable
+
+      AnonymousCommand = Class.new(Dry::CLI::Command)
+
+      delegate %i[desc example argument option] => AnonymousCommand
+
+      def run(arguments: ARGV, out: $stdout)
+        command = AnonymousCommand
+        command.define_method(:call) do |*args|
+          yield(*args)
+        end
+
+        Dry.CLI(command).call(arguments: arguments, out: out)
+      end
+    end
+  end
+end
+
+# `dry/cli/inline` is not required by default
+# and explicit requirement of this file means that
+# it is expected of abusing global namespace with
+# methods above
+#
+# Example:
+#   require 'bundler/inline'
+#   gemfile { gem 'dry/cli', require: 'dry/cli/inline' }
+#
+#   desc 'List files in a directory'
+#   argument :path, required: false, desc: '[DIR]'
+#   option :all, aliases: ['a'], type: :boolean
+#
+#   run do |path: '.', **options|
+#     puts options.key?(:all) ? Dir.entries(path) : Dir.children(path)
+#   end
```

Comment:
I think this doc section should be right above `Inline` module definition, otherwise yard won't pick it up correctly